lib/src/model/game/material_diff.dart

```diff
@@ -35,6 +36,29 @@ class MaterialDiff with _$MaterialDiff {
     final IMap<Role, int> blackCount = board.materialCount(Side.black);
     final IMap<Role, int> whiteCount = board.materialCount(Side.white);
 
+    final IMap<Role, int> blackStartingCount =
+        Board.standard.materialCount(Side.black);
+    final IMap<Role, int> whiteStartingCount =
+        Board.standard.materialCount(Side.white);
+    // TODO: parameterise starting position maybe so it can be passed in
+
+    IMap<Role, int> subtractPieceCounts(
+      IMap<Role, int> startingCount,
+      IMap<Role, int> subtractCount,
+    ) {
+      IMap<Role, int> capturedPieces = IMap();
```

Comment:
This could be written in a pure functional way:

```
return startingCount.map(
    (role, count) => MapEntry(role, count - (subtractCount.get(role) ?? 0)),
);

```

Reply:
Thanks, I thought this should be possible but I couldn't figure out the right way to do it